client/lib/posts/post-edit-store.js

```diff
@@ -471,7 +471,23 @@ PostEditStore = {
 	},
 
 	isDirty: function() {
-		return _post !== _savedPost || _rawContent !== _initialRawContent;
+		debug(
+			'isDirty post?%s rawContent?%s initial=%s content=%s',
+			_post !== _savedPost,
+			_rawContent !== _initialRawContent,
+			_initialRawContent,
+			_rawContent
+		);
+		if ( _post !== _savedPost ) {
+			return true;
+		}
+		if ( _rawContent !== _initialRawContent ) {
+			return (
+				! isContentEmpty( _rawContent ) ||
```

Comment:
My thinking was to prevent raw content from being set as empty before allowing it to get to this point. Especially since `isDirty` is run alot.

```diff
diff --git a/client/lib/posts/post-edit-store.js b/client/lib/posts/post-edit-store.js
index a8c695e..4777058 100644
--- a/client/lib/posts/post-edit-store.js
+++ b/client/lib/posts/post-edit-store.js
@@ -237,6 +237,8 @@ function normalize( post ) {
 function setRawContent( content ) {
        var isDirty, hasContent;
 
+       content = content.replace( REGEXP_EMPTY_CONTENT, '' );
+
        if ( null === _initialRawContent ) {
                debug( 'Set initial raw content to: %s', content );
                _initialRawContent = content;
```